src/main/scala/ai/privado/utility/ImportUtility.scala

```diff
@@ -0,0 +1,102 @@
+            case "java" => "(import)\\s+" // Almost always is accompanied by a space at the end
+            case "py" => "(import)\\s+" // Almost always is accompanied by a space at the end
+            case default => "(import)\\s+" // Default is java 
+        }
+        return result;
+    }
+
+    private def returnImportStatements(filePath: String, language: String): Set[String] = {
+        val source = Source.fromFile(filePath)
+        val matchImportRegex = getMatchImportRegex(language);
+        val onlyLiteralWordRegex = getLiteralWordRegex(language);
+
+        var multilineFlag = false;
+        var uniqueImports: Set[String] = Set()
+        val result = new StringBuilder("")
+        try {
+            breakable
+            {
+                for (line <- source.getLines()) {
+                        val scanLine = line.trim();
+                        if (scanLine.matches("/\\*.*")) { // set flag if multiline comment is encountered
+                            multilineFlag = true;
+                        }    
+
+                        // Ignore if inside a multiline comment
+                        if (!multilineFlag) {
+                            if (scanLine matches matchImportRegex) {
+                                val withoutImportLine = scanLine.replace(onlyLiteralWordRegex.r.findFirstIn(line).mkString, "");
+                                uniqueImports += withoutImportLine; // add each import to set
+                            } else {
+                                if (!scanLine.matches("(package|//)\\s*.*") && scanLine.lengthCompare(0) != 0) { // Ignore if there is nothing or a package definition on a line
+                                    break
+                                }
+                            }
+                        }
+
+                        if (scanLine.matches(".*\\*/")) {
+                            multilineFlag = false;
+                        }
+                        
+                }
+            }
+        } catch {
+            case e: MalformedInputException => println(e)
+        }
+        
+        source.close()
+        return uniqueImports;
+    }
+
+    private def scanAllFilesInDirectory(dirpath: String): Unit = {
+        val files = new java.io.File(dirpath).listFiles();
+        for (file <- files) {
+            if (file.isDirectory()) scanAllFilesInDirectory(file.getAbsolutePath()) // Recursively call for each directory
+            else {
+                if (file.getAbsolutePath().matches(".+.(java)")) { // Scan only java files
+                    // println(file.getAbsolutePath())
+                    var fileImports: Set[String] = returnImportStatements(file.getAbsolutePath(), "java");
+                    for (el <- fileImports) {
+                        allImports += el;
+                    }
+                }
+            }
+        }
+
+    }
+
+    def getAllImportsFromDirectory(dirpath: String): Set[String] = {
```

Comment:
Add the Detected Language as part of input to this method. This will ensure only the files relevant to us are getting scanned based on the language engine used.